src/java.base/share/classes/java/io/File.java

```diff
@@ -1376,7 +1376,11 @@ public boolean mkdirs() {
      * file from one filesystem to another, it might not be atomic, and it
      * might not succeed if a file with the destination abstract pathname
      * already exists.  The return value should always be checked to make sure
-     * that the rename operation was successful.
+     * that the rename operation was successful.  As instances of {@code File}
+     * are immutable, the abstract pathname represented by this {@code File}
+     * object does not itself change although the filesystem object it denoted
```

Comment:
I guess you meant `file` object here, instead of `filesystem`?

Reply:
No, I intended `filesystem` but in the sense of "object in the filesystem" but it does seem awkward.

Reply:
It might be clearer if the end of the sentence were changed to something like "... this File object is not changed to name destination file or directory".

Reply:
Like so?
```
--- a/src/java.base/share/classes/java/io/File.java
+++ b/src/java.base/share/classes/java/io/File.java
@@ -1376,7 +1376,9 @@ public class File
      * file from one filesystem to another, it might not be atomic, and it
      * might not succeed if a file with the destination abstract pathname
      * already exists.  The return value should always be checked to make sure
-     * that the rename operation was successful.
+     * that the rename operation was successful.  As instances of {@code File}
+     * are immutable, this File object is not changed to name the destination
+     * file or directory.
      *
      * <p> Note that the {@link java.nio.file.Files} class defines the {@link
      * java.nio.file.Files#move move} method to move or rename a file in a
```

Reply:
yes, I think that works.

Reply:
[CSR](https://bugs.openjdk.java.net/browse/JDK-8261935) also so updated.